### DISCUSSION TITLE
Missing new event

### DISCUSSION THREAD

**Discussion by diegargon** [blakeblackshear/frigate#14036] [Ask A Question] [answered]
Hello.

In this scenario + yellow zone:

https://imgur.com/a/HiQIhdQ

what can cause the missing 'new' event? 

when an object cross up->down or down->up sometimes the 'new' event is missing, only got few 'updates' and 'end' tagged with zone.

**Comment by NickM-27** [comment]
the new event will always be sent first

**Reply by diegargon** [reply]
I suppose... but what can be cause the missing new event?

I have configured two automation in HA

One send the image when new+zone is detected

The other send a message with label+zone+type for debug purpose

Mostly the new event is triggered but sometimes i receive only the debug messages; 
person+zone+update
person+zone+update
person+zone+end

Mostly this occurs when a car/person go outside (down->up in the image)

**Reply by NickM-27** [reply]
it is most likely just an error in your automation, there is no guarantee that the object will be in the zone during the new message. Nothing has changed with the handling of these MQTT messages in many frigate versions.

**Reply by diegargon** [reply]
> Nothing has changed with the handling of these MQTT messages in many frigate versions.

New user.

>  there is no guarantee that the object will be in the zone during the new message. 

that mean the new event is not triggered when a object entered in a zone?

It seems that I simplle not understood how events works (Even after reading the docs)

**Reply by NickM-27** [reply]
`/events` is a raw topic, it represents objects that are being tracked. a `new` event is sent when the object is first detected as a true positive

**Reply by diegargon** [reply]
ok that why then

Lots (if not all)  HA automation examples in the HA forum use new+zone for trigger the message system. (since update is triggered few times)

According to what you say, this is probably not right for most uses that you want to alert when an object enters the zone.

**Reply by NickM-27** [reply, marked as answer]
right, I would always recommend just checking the `before` and `after` zones so you trigger a notification on the first time the object entered the zone

**Reply by diegargon** [reply]
the problem is the before=zone after=zone is triggered lots of times with the update event causing lots of messages.

I have to think the best way to do that, since the only uniq events are new and end + check zone, and new like you say no always is triggered with a zone and end will delay it in some scenarios. (and probably the same as new, is not guarantee that have the zone in after/before))

**Reply by NickM-27** [reply]
> the problem is the before=zone after=zone is triggered lots of times with the update event causing lots of messages.

no, it can't be. If you compare before and after zones and only send a notification when before zones is empty and after zones is not empty you will only have the possibility of getting a notification one time per object.

**Reply by diegargon** [reply]
i will check that thanks
